usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -1567,7 +1569,7 @@ arc_cksum_is_equal(arc_buf_hdr_t *hdr, zio_t *zio)
 			bzero((char *)cbuf + csize, HDR_GET_PSIZE(hdr) - csize);
```

Comment:
BTW, this wasn't part of your change but this also looks broken. I believe that we should be using `abd_zero_off` here rather than bzero.

Reply:
ok. will do.